HARK/distribution.py

```diff
@@ -281,58 +305,68 @@ class Lognormal(Distribution):
+        sigma : Union[float, np.ndarray], optional
+            Standard deviation of underlying normal distribution, by default 1.0
+        seed : Optional[int], optional
+            Seed for random number generator, by default None
+
+        Returns
+        -------
+        Lognormal or DiscreteDistribution
+            Lognormal distribution or DiscreteDistribution with a single atom.
+        """
+
+        if sigma == 0:
+            # If sigma is zero, return a DiscreteDistribution with a single atom
+            return DiscreteDistribution([1.0], [np.exp(mu)], seed=seed)
+
+        return super(Lognormal, cls).__new__(cls)
+
+    def __init__(
+        self,
+        mu: Union[float, np.ndarray] = 0.0,
+        sigma: Union[float, np.ndarray] = 1.0,
+        seed: Optional[int] = 0,
+    ):
+        self.mu = np.asarray(mu)
+        self.sigma = np.asarray(sigma)
 
         if self.mu.size != self.sigma.size:
-            raise Exception(
+            raise AttributeError(
                 "mu and sigma must be of same size, are %s, %s"
                 % ((self.mu.size), (self.sigma.size))
             )
 
-    def draw(self, N):
-        """
-        Generate arrays of lognormal draws. The sigma parameter can be a number
-        or list-like.  If a number, output is a length N array of draws from the
-        lognormal distribution with standard deviation sigma. If a list, output is
-        a length T list whose t-th entry is a length N array of draws from the
-        lognormal with standard deviation sigma[t].
-
-        Parameters
-        ----------
-        N : int
-            Number of draws in each row.
-
-        Returns:
-        ------------
-        draws : np.array or [np.array]
-            T-length list of arrays of mean one lognormal draws each of size N, or
-            a single array of size N (if sigma is a scalar).
-        """
-
-        draws = []
-        for j in range(self.mu.size):
-            draws.append(
-                self.RNG.lognormal(
-                    mean=self.mu.item(j), sigma=self.sigma.item(j), size=N
-                )
-            )
-        # TODO: change return type to np.array?
-        return draws[0] if len(draws) == 1 else draws
+        # Set up the RNG
+        super().__init__(
+            stats.lognorm, s=self.sigma, scale=np.exp(self.mu), loc=0, seed=seed
+        )
 
-    def approx(self, N, tail_N=0, tail_bound=None, tail_order=np.e):
+    def _approx_equiprobable(
+        self, N, endpoints=False, tail_N=0, tail_bound=None, tail_order=np.e
+    ):
         """
         Construct a discrete approximation to a lognormal distribution with underlying
         normal distribution N(mu,sigma).  Makes an equiprobable distribution by
         default, but user can optionally request augmented tails with exponentially
         sized point masses.  This can improve solution accuracy in some models.
 
+        TODO: add endpoints option
```

Comment:
Is there an issue for this in the issue tracker?

Are you going to do it in this PR, or leave it to a later release?

Reply:
I was thinking of leaving it for a different PR since I still need to discuss how to do this with the group. 

I will make an issue for this.

Reply:
#1205